include/view.h

```diff
@@ -160,6 +160,66 @@ struct xdg_toplevel_view {
 	struct wl_listener new_popup;
 };
 
+enum lab_view_criteria {
+	LAB_VIEW_CRITERIA_NONE = 0,
+	LAB_VIEW_CRITERIA_CURRENT_WORKSPACE = 1 << 0,
+	LAB_VIEW_CRITERIA_NO_ALWAYS_ON_TOP = 1 << 1,
+	LAB_VIEW_CRITERIA_NO_SKIP_WINDOW_SWITCHER = 1 << 2,
+};
+
+/**
+ * for_each_view() - iterate over all views which match criteria
+ * @view: Iterator.
+ * @head: Head of list to iterate over.
+ * @criteria: Criteria to match against.
+ * Example:
+ *	struct view *view;
+ *	for_each_view(view, &server->views, LAB_VIEW_CRITERIA_NONE) {
+ *		printf("%s\n", view_get_string_prop(view, "app_id"));
+ *	}
+ */
+#define for_each_view(view, head, criteria)			\
+	for (view = view_next(head, NULL, criteria); view;	\
+		view = view_next(head, view, criteria))
```

Comment:
My OCD kicks in with "indent with tabs align with spaces" and

```C
	for (view = view_next(head, NULL, criteria);
	     view;
	     view = view_next(head, view, criteria))
```

IMO it would be better, but this is purely my style suggestions.

Reply:
Well you know... we don't normally align with spaces (Devault coding style) - but checkpatch.pl does not complain and it does look nicer :smile:
Implemented.